dissect/target/tools/shell.py

```diff
@@ -1331,29 +1331,39 @@ def build_pipe_stdout(pipe_parts: list[str]) -> Iterator[TextIO]:
         yield pipe_stdin
 
 
-def open_shell(targets: list[str | pathlib.Path], python: bool, registry: bool) -> None:
+def open_shell(targets: list[str | pathlib.Path], python: bool, registry: bool, commands: list[str]) -> None:
     """Helper method for starting a regular, Python or registry shell for one or multiple targets."""
     targets = list(Target.open_all(targets))
 
     if python:
-        python_shell(targets)
+        python_shell(targets, commands=commands)
     else:
         cli_cls = RegistryCli if registry else TargetCli
-        target_shell(targets, cli_cls=cli_cls)
+        target_shell(targets, cli_cls=cli_cls, commands=commands)
 
 
-def target_shell(targets: list[Target], cli_cls: type[TargetCmd]) -> None:
+def target_shell(targets: list[Target], cli_cls: type[TargetCmd], commands: list[str]) -> None:
     """Helper method for starting a :class:`TargetCli` or :class:`TargetHubCli` for one or multiple targets."""
     if cli := create_cli(targets, cli_cls):
+        if commands is not None:
+            # @todo if cli is not RegistryCli: - load targetrc
```

Comment:
Be advised: RegistryCli also supports a targetrc file

Reply:
Addendum: Since bash does **not** execute the run commands file for non-interactive shells, you can skip loading the targetrc file, until someone complaints. (discussed with @pyrco)

Reply:
So this means, this PR does not have to wait for targetrc and does not have to integrate with targetrc?

Reply:
Yes.